operator/pkg/client/schemas/syncer_test.go

```diff
@@ -0,0 +1,158 @@
+	schema.Status.SchemaHash = hash
+}
+
+func expectSchemasMatch(t *testing.T, ctx context.Context, syncer *Syncer, schema *v1alpha2.Schema) {
+	normalizeSchema(t, ctx, syncer, schema)
+
+	expectedSchema, err := schemaFromV1Alpha2Schema(schema)
+	require.NoError(t, err)
+
+	actualSchema, err := syncer.getLatest(ctx, schema)
+	require.NoError(t, err)
+
+	require.True(t, expectedSchema.CompatibilityEquals(actualSchema), "Compatibility levels not equal %+v != %+v", actualSchema.CompatibilityLevel, expectedSchema.CompatibilityLevel)
+	require.Equal(t, expectedSchema.Schema, actualSchema.Schema)
+	require.True(t, expectedSchema.SchemaEquals(actualSchema), "Schemas not equal %+v != %+v", actualSchema, expectedSchema)
+}
+
+func expectSchemaUpdate(t *testing.T, ctx context.Context, syncer *Syncer, schema *v1alpha2.Schema, update bool) {
+	t.Helper()
+
+	_, versions, err := syncer.Sync(ctx, schema)
+	require.NoError(t, err)
+
+	if !update {
+		require.EqualValues(t, schema.Status.Versions, versions)
+	} else {
+		require.Len(t, versions, len(schema.Status.Versions)+1, "update expected, but didn't create another schema version")
+		schema.Status.Versions = versions
+	}
+
+	expectSchemasMatch(t, ctx, syncer, schema)
+
+	if update {
+		// check to make sure we don't update again
+		expectSchemaUpdate(t, ctx, syncer, schema, false)
+	}
+}
+
+func TestSyncer(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*2)
+	defer cancel()
+
+	container, err := redpanda.Run(ctx, "docker.redpanda.com/redpandadata/redpanda:v23.2.8",
+		redpanda.WithEnableSchemaRegistryHTTPBasicAuth(),
+		redpanda.WithEnableKafkaAuthorization(),
+		redpanda.WithEnableSASL(),
+		redpanda.WithSuperusers("user"),
+		redpanda.WithNewServiceAccount("user", "password"),
+	)
+
+	require.NoError(t, err)
+
+	schemaRegistry, err := container.SchemaRegistryAddress(ctx)
+	require.NoError(t, err)
+
+	schemaRegistryClient, err := sr.NewClient(sr.BasicAuth("user", "password"), sr.URLs(schemaRegistry))
+	require.NoError(t, err)
+
+	syncer := NewSyncer(schemaRegistryClient)
+
+	schema := &v1alpha2.Schema{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "schema",
+		},
+		Spec: v1alpha2.SchemaSpec{
+			Text: validSchema,
+		},
+	}
+
+	reference := &v1alpha2.Schema{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "reference",
+		},
+		Spec: v1alpha2.SchemaSpec{
+			Text: validSchema,
+		},
+	}
+
+	// create initial schema and reference
+	expectSchemaUpdate(t, ctx, syncer, schema, true)
+	expectSchemaUpdate(t, ctx, syncer, reference, true)
+
+	// update references
+	schema.Spec.References = []v1alpha2.SchemaReference{
+		{
+			Subject: reference.Name,
+			Name:    "test",
+			Version: 1,
+		},
+	}
+	expectSchemaUpdate(t, ctx, syncer, schema, true)
+
+	// update compatibility level
+	schema.Spec.CompatibilityLevel = ptr.To(v1alpha2.CompatabilityLevelFull)
+	expectSchemaUpdate(t, ctx, syncer, schema, false)
+
+	// TODO: Request from core support for the following"
```

Comment:
Should we file these as tickets and link back to them or are they far enough down the priority list that we're not too worried?

Reply:
Added a comment linking to the ticket I had already filed for schema rules and metadata, I'll file the other ones as a follow-up.